### DISCUSSION TITLE
Deployment Question

### DISCUSSION THREAD

**Discussion by pscemama-mitre** [usnistgov/dioptra#454] [General]
I'm running into the following

```bash
 ✘ nginx-ssl Error                                                                                                                                        0.1s 
Error response from daemon: pull access denied for dioptra/nginx, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

```bash
./init-deployment.sh --branch dev
```

From [these directions](https://github.com/usnistgov/dioptra/tree/dev/cookiecutter-templates/cookiecutter-dioptra-deployment/%7B%7Bcookiecutter.__project_slug%7D%7D). 

I'm sure I did something wrong with my configuration, but I can't seem to figure out how to print out the configuration to check things. Any ideas @jkglasbrenner @keithmanville ? Thanks!

**Comment by jkglasbrenner** [comment]
@pscemama-mitre That usually happens to me when I forget to build one of the images. Run `docker images`, if you don't see dioptra/nginx listed then you need to build it first. If you're running `make build-nginx` in your cloned `dioptra` folder and it's saying that there's nothing to do, then run `make clean` first and try again.

**Reply by pscemama-mitre** [reply]
It is not listed, so I have to build it. How do I build the image?

**Reply by chisholm** [reply]
cd into your dioptra clone directory, and try running `make build-nginx`.

**Reply by pscemama-mitre** [reply]
That worked thanks! 

Now running into this


```bash
Error response from daemon: pull access denied for dioptra/mlflow-tracking, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

When running

```bash
./init-deployment.sh --branch dev
```

From [these directions](https://github.com/usnistgov/dioptra/tree/dev/cookiecutter-templates/cookiecutter-dioptra-deployment/%7B%7Bcookiecutter.__project_slug%7D%7D).

**Reply by jkglasbrenner** [reply]
It's the same error, just for a different container. Make sure you've built the 5 containers as documented here before attempting to initialize your deployment: https://pages.nist.gov/dioptra/getting-started/building-the-containers.html. You can skip the GPU builds for PyTorch and Tensorflow.

**Reply by pscemama-mitre** [reply]
Sounds good. Got that to work but both the pytorch-gpu and pytorch-cpu builds cause errors, so I just did tensorflow2-cpu.

Now I am running into this error:

```bash
Copy Extra CA Certificates: INFO - No extra CA files provided, will use default CA bundle
Git Clone: INFO - Cloning https://github.com/usnistgov/dioptra.git to /volumes/init-repos (dev branch)
Git Clone: WARNING - No personal access token provided, clone will fail if https://github.com/usnistgov/dioptra.git is not public.
Git Clone: INFO - Executing git clone -b dev --single-branch https://github.com/usnistgov/dioptra.git
Cloning into 'dioptra'...
fatal: unable to access 'https://github.com/usnistgov/dioptra.git/': server certificate verification failed. CAfile: none CRLfile: none
```

I assume I have to configure my github personal access token? How would I do that?

**Reply by jkglasbrenner** [reply]
That's a corporate certificates issue, you need to download the CA certificates and put them in the docker/ca-certificates folder. See this README: https://github.com/usnistgov/dioptra/blob/main/docker/ca-certificates/README.md. You also need to include them in ssl/ca-certificates folder of your deployment folder (the folders created using the cruft tool).

If you need more specific details, we can follow-up outside this forum.

**Reply by pscemama-mitre** [reply]
@jkglasbrenner yeah let me know if you can jump on a call this afternoon or tomorrow? I copied the certificate files over to the docker/ca-certificates directory but `'No extra CA files provided, will use default CA bundle'` when trying to initialize the deployment, thus leading to the same error.

**Comment by chisholm** [comment]
There's a different place to copy certificate files to, when deploying.  Try copying them to `<your-deployment-directory>/ssl/ca-certificates`.

**Reply by pscemama-mitre** [reply]
Thanks! This worked and I was able to deploy.
